src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -346,5 +353,52 @@ private BoundExpression RewriteStringConcatInExpressionLambda(CSharpSyntaxNode s
 
             return new BoundBinaryOperator(syntax, operatorKind, loweredLeft, loweredRight, default(ConstantValue), method, default(LookupResultKind), type);
         }
+
+        /// <summary>
+        /// Checks whether the expression represents a boxing conversion of a value type.
+        /// If it does, it returns a synthesized ToString call for that value type instance.
+        /// Otherwise, it returns the original expression.
+        /// </summary>
+        private BoundExpression ConvertExpressionToStringIfValueType(CSharpSyntaxNode syntax, BoundExpression expr)
+        {
+            // The lowered expressions coming in will be boxing conversions if the input is a value type.
+            // When passed to String.Concat, the resulting object will have ToString called on it.
+            // We can optimize away the boxing allocation by just calling ToString on the value type directly.
```

Comment:
Good point.  The majority of the cases I've seen this occur with are in fact with chars, ints, byte, and just a handful of other core types, so I think the decrease in value for the change would be fairly negligable.  If you agree this is worthwhile to pursue, I'll go ahead and make that change...?